### ISSUE TITLE
Unhelpful error message for converting RDKit molecule without metadata

### ISSUE THREAD

**Issue by jbarnoud** [MDAnalysis/mdanalysis#3318]
## Expected behavior ##

A user can convert an atom group to rdkit, use rdkit to add hydrogens, and load the modified molecule back in MDAnalysis.

## Actual behavior ##

The universe fails to be created.

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
import rdkit

u = mda.fetch_mmtf('3cs9')
nilotinib = u.segments[4].atoms

# MMTF does not fill the elements attribute
u.add_TopologyAttr('elements', u.atoms.types)

nilotinib_noh = nilotinib.convert_to.rdkit(NoImplicit=False)
nilotinib_h = rdkit.Chem.AddHs(nilotinib_noh)
# Getting the 3d structure is not necessary to reproduce
# the issue but it is part of the workflow
# rdkit.Chem.AllChem.EmbedMolecule(nilotinib_h)
nilotinib_u = mda.Universe(nilotinib_h)
```

This raises the following exception:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
c:\users\dr19502\dev\mdanalysis\package\MDAnalysis\core\universe.py in _topology_from_file_like(topology_file, topology_format, **kwargs)
    121         with parser(topology_file) as p:
--> 122             topology = p.parse(**kwargs)
    123     except (IOError, OSError) as err:

c:\users\dr19502\dev\mdanalysis\package\MDAnalysis\converters\RDKitParser.py in parse(self, **kwargs)
    331                        atom_resindex=residx,
--> 332                        residue_segindex=segidx)
    333 

c:\users\dr19502\dev\mdanalysis\package\MDAnalysis\core\topology.py in __init__(self, n_atoms, n_res, n_seg, attrs, atom_resindex, residue_segindex)
    472                              atom_resindex=atom_resindex,
--> 473                              residue_segindex=residue_segindex)
    474 

c:\users\dr19502\dev\mdanalysis\package\MDAnalysis\core\topology.py in __init__(self, n_atoms, n_residues, n_segments, atom_resindex, residue_segindex)
    219             if not len(self._AR) == n_atoms:
--> 220                 raise ValueError("atom_resindex must be len n_atoms")
    221         self._RA = make_downshift_arrays(self._AR, n_residues)

ValueError: atom_resindex must be len n_atoms

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-88-609512996fb5> in <module>
      9 # the issue but it is part of the workflow
     10 #rdkit.Chem.AllChem.EmbedMolecule(nilotinib_h)
---> 11 nilotinib_u = mda.Universe(nilotinib_h)

c:\users\dr19502\dev\mdanalysis\package\MDAnalysis\core\universe.py in __init__(self, topology, all_coordinates, format, topology_format, transformations, guess_bonds, vdwradii, in_memory, in_memory_step, *coordinates, **kwargs)
    336             topology = _topology_from_file_like(self.filename,
    337                                                 topology_format=topology_format,
--> 338                                                 **kwargs)
    339 
    340         if topology is not None:

c:\users\dr19502\dev\mdanalysis\package\MDAnalysis\core\universe.py in _topology_from_file_like(topology_file, topology_format, **kwargs)
    138             "Failed to construct topology from file {0}"
    139             " with parser {1}.\n"
--> 140             "Error: {2}".format(topology_file, parser, err))
    141     return topology
    142 

ValueError: Failed to construct topology from file <rdkit.Chem.rdchem.Mol object at 0x00000220A76D7B70> with parser <class 'MDAnalysis.converters.RDKitParser.RDKitParser'>.
Error: atom_resindex must be len n_atoms
```

My understanding is that the issues is due to the added hydrogens not having all the metadata the other atoms have. This leads to several precursor list to topology attribute to be populated only with the atoms from the original structure in the conditional at https://github.com/MDAnalysis/mdanalysis/blob/acc713c272bcb33d9047fc30d6d0c2d578767366/package/MDAnalysis/converters/RDKitParser.py#L203

## Current version of MDAnalysis ##

- Which version are you using? '2.0.0-dev0'
- Which version of Python (`python -V`)? '3.7.6 | packaged by conda-forge | (default, Jan  7 2020, 21:48:41) [MSC v.1916 64 bit (AMD64)]'
- Which operating system? Windows

**Comment by cbouy** [comment]
Can you try the following? 
```diff
- nilotinib_h = Chem.AddHs(nilotinib_noh)
- AllChem.EmbedMolecule(nilotinib_h) 
+ nilotinib_h = Chem.AddHs(nilotinib_noh, addResidueInfo=True, addCoords=True)
```
This should fix the issue and add 3D coordinates to the hydrogens at the same time

PS: in your example you forgot `nilotinib = u.select_atoms(... )`

**Comment by jbarnoud** [comment]
> Can you try the following?

It works!

> PS: in your example you forgot nilotinib = u.select_atoms(... )

Thanks, I added it to my original post.

So the issue now is that the universe creation fails without a useful message. Less of a problem.
